### DISCUSSION TITLE
License Change: Keep GPLv3 or Change? (feedback request)

### DISCUSSION THREAD

**Discussion by outrolled** [serghey-rodin/vesta#2300] [General]
### Please provide your feedback for this topic, this is a very important question

Given that we are rewriting Vesta from scratch, we have the rare opportunity to change the project licensing. I am a defender of GPLv3, however I am open to other licenses with the following goals:
- Keep VestaCP core product Open Source and free
- Prevent competing forks from diluting resources (creating incentives for more people to contribute to Vesta)
- Ensure that we get enough contributors OR we can fund paid developers
- Maybe limit competition for a few years until the code becomes open again (like the BSL license allows, I think its a bad idea)

There are a few options for licensing, some licenses we looked at were:
- Keeping GPLv3
- AGPL 3.0 (my preferred option)
- BSL 1.1
- Functional Source License (FSL)
- Something else?

I personally believe that **AGPLv3** may be the best option as we keep the same level of permissions, but we plug the hole of network distribution in GPLv3 among other things. AGPL license is recommended by Free Software Foundation (FSF) if the software is commonly run on the cloud (which Vesta is).

Given that our monetization/funding strategy is via paid plugins, the core product should stay as open as possible. We will be able to sell plugins for managing paid hosting (similar to WHMCS) that will come with commercial licenses that allow you to make money from VestaCP as well.

What do you think about AGPL 3.0?

**Comment by Danter3** [comment]
Why should the license be changed? First of all, I'm not a fan of changing licenses for well-established software. There will always be resources all over the internet stating that VestaCP is GPLv3 licensed, which will inevitably cause confusion and misunderstandings. Obviously, the logical step for something like this would be to launch new software, but you want the benefits of established software like VestaCP without the ethical limitations that come with it.

Regarding the reason for forks, is that even a good thing? With all the WordPress drama, what would be the benefit of that in the face of erratic behavior like Matt's? I mean, forks only arise if the software is abandoned and its maintainers aren't willing to merge PRs, or if there are significant disagreements in the community about development or ethics. If that's not the case, there shouldn't be reasons to create forks.

I saw on the website: "For hosters: We have special licensing options for hosting companies, just contact us to chat."

I guess this means the AGPL 3.0 license is already a done deal, which I see as a mistake because it basically impacts a lot of edge cases, like the internal use of the panel for a network, etc. I wish I had seen this months ago... 🤦

**Comment by outrolled** [comment]
Hi @Danter3 the reason that we are reconsidering a license change is because Vesta is being re-written from the ground up, so this is a rare opportunity to rethink our license. I wouldn't say that Vesta is a well established software anymore - it is in real need of love, a more secure architecture, more robust systems, etc.

I am also considering a license change as I am personally investing a lot of my own money to develop vesta, had zero external contributions so far, and I am concerned that after all of this investment we can't maintain the project well. I really want to make Vesta sustainable and well maintained.

Please don't assume that we will do what Wordpress/Matt did, we are committed to open source, and our monetization strategy is focused on paid plugins for features that are not part of the Vesta core. We made the file manager plugin free and gave a free license to other forks as well. Vesta will always be free, and the only paid features will be things like payments and client management (which Vesta never had) with similar functionality to WHMCS (hence the link for hosters on the new site).

I agree that forks mostly happen when there are misalignments, however I've seen many cases where disagreements around architecture or features can severely dilute a project (contributors split, the main project suffers, etc). Large OSS projects are generally immune, but small ones like Vesta are quite vulnerable in my opinion.

I appreciate your comment, as no one provided any thoughts on this for a long time. I can confirm that the AGPL license is not a done thing, and I am still open to other alternatives and views until Vesta 2.0 is published. I agree that AGPL creates weird edge cases, and it could be interpreted as very limiting or viral. I honestly wish there was an easy answer to this one - I am quite open to some new and less popular licenses as well.

PS.  The current version of vesta is quite outdated, so please feel free to check out some of the other forks that are better maintained and are more secure at the moment. Vesta 2.0 will be worth the wait, but we are really behind the schedule at the moment.

**Comment by Neustradamus** [comment]
@outrolled has done a new post here:
- https://github.com/outroll/vesta/discussions/2315

Linked to:
- https://github.com/outroll/vesta/issues/2280
- https://github.com/outroll/vesta/discussions/2296
- https://github.com/outroll/vesta/discussions/2298
- https://github.com/outroll/vesta/discussions/2299
- https://github.com/outroll/vesta/discussions/2300
- https://github.com/outroll/vesta/issues/2312
- https://github.com/outroll/vesta/discussions/2315
- https://github.com/outroll/vesta/issues/2321
